# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/perfmonitoring/PerformanceUtils.cpp
# Commit: c0849928f934
# Full Hash: c0849928f934f0b5e931813eaadc23f0fb8521d7
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-23 09:52:48
# Regressor Bug: 1580565
# File Overlap Count: 2
# Description:
#   Bug 1580565 - Part 6: Add a unique ID to each BrowsingContextGroup, r=kmag
#   
#   This allows us to explicitly specify BrowsingContextGroups when synchronizing
#   them. A major advantage of this is that it means we can handle an attempt to
#   create a BrowsingContext with a parent which the content process is unaware of,
# ==============================================================================

diff -r 3d4f155096be -r c0849928f934 toolkit/components/perfmonitoring/PerformanceUtils.cpp
--- a/toolkit/components/perfmonitoring/PerformanceUtils.cpp	Wed Apr 22 15:48:15 2020 +0000
+++ b/toolkit/components/perfmonitoring/PerformanceUtils.cpp	Wed Apr 22 16:25:18 2020 +0000
@@ -41,12 +41,7 @@
   }
 
   nsTArray<RefPtr<BrowsingContextGroup>> groups;
-  if (XRE_IsContentProcess()) {
-    groups.AppendElements(
-        ContentChild::GetSingleton()->BrowsingContextGroups());
-  } else {
-    groups.AppendElements(ContentParent::BrowsingContextGroups());
-  }
+  BrowsingContextGroup::GetAllGroups(groups);
 
   nsTArray<DocGroup*> docGroups;
   for (auto& browsingContextGroup : groups) {